src/json_stream/tokenizer/strings.py

```diff
@@ -0,0 +1,169 @@
+import io
+import unicodedata
+from typing import Union
+from io import DEFAULT_BUFFER_SIZE
+
+STRING_ESCAPE_CODES = {
+    '\\': '\\',
+    '/': '/',
+    '"': '"',
+    'b': '\b',
+    'f': '\f',
+    'n': '\n',
+    't': '\t',
+    'r': '\r'
+}
+
+SURROGATE = 'Cs'
+
+CHAR = 1
+STRING_ESCAPE = 2
+UNICODE = 4
+UNICODE_SURROGATE_START = 5
+UNICODE_SURROGATE_STRING_ESCAPE = 6
+UNICODE_SURROGATE = 7
+
+
+class JsonStringReader(io.TextIOBase):
+    def __init__(self, stream: io.TextIOBase, initial_buffer=''):
+        self.stream = stream
+        self.buffer = initial_buffer
+        self.readline_buffer = ''
+        self.unicode_buffer = ''
+        self.state = CHAR
+        self.end_of_string = False
+        self.index = 0
+
+    @property
+    def complete(self):
+        return self.end_of_string and not self.readline_buffer
+
+    def readable(self) -> bool:
+        return True
+
+    def read(self, size: Union[int, None] = None) -> str:
+        result = ''
+        length = DEFAULT_BUFFER_SIZE
+        while not self.complete and (size is None or not result):
+            if size:
+                length = size - len(result)
+            result += self._read_chunk(length)
+        return result
+
+    def _read_chunk(self, size: int) -> str:
+        if self.readline_buffer:
+            result, self.readline_buffer = self.readline_buffer[:size], self.readline_buffer[size:]
+            return result
+        chunk = self.buffer or self.stream.read(size)
```

Comment:
I've noticed a new issue here. When the buffering argument is passed, the string reader still reads DEFAULT_BUFFER_SIZE from the stream. Working on a fix...